lib/std/jsonutils.nim

```diff
@@ -92,31 +98,59 @@ proc checkJsonImpl(cond: bool, condStr: string, msg = "") =
 template checkJson(cond: untyped, msg = "") =
   checkJsonImpl(cond, astToStr(cond), msg)
 
-template fromJsonFields(a, b, T, keys) =
-  checkJson b.kind == JObject, $(b.kind) # we could customize whether to allow JNull
+proc hasField[T](obj: T, field: string): bool =
+  for k, _ in fieldPairs(obj):
+    if k == field:
+      return true
+  return false
+
+macro accessField(obj: typed, name: static string): untyped = 
+  newDotExpr(obj, ident(name))
+
+template fromJsonFields(newObj, oldObj, json, discKeys, numMatched, opt) =
+  type T = typeof(newObj)
+  # we could customize whether to allow JNull
+  checkJson json.kind == JObject, $json.kind
   var num = 0
-  for key, val in fieldPairs(a):
+  for key, val in fieldPairs(newObj):
     num.inc
-    when key notin keys:
-      if b.hasKey key:
-        fromJson(val, b[key])
+    when key notin discKeys:
+      if json.hasKey key:
+        numMatched.inc
+        fromJson(val, json[key])
+      elif opt.allowMissingKeys:
+        # if there is no discriminant keys the `oldObj` must always have the
+        # same keys and the new one. Otherwise we must check, because the could
```

Comment:
s/and the/as the/

Reply:
@timotheecour you can use the GitHub suggest feature btw :)

Reply:
Fixed.

Reply:
@Yardanico true true... on the fence